tasks/deploy.py

```
@@ -0,0 +1,445 @@
+            if re_targz_created.match(line):
+                # tarball created
+                targz_created = True
+
+    # we test results from the above check and if there is one tarball only
+    if no_missing_modules and targz_created and len(eessi_tarballs) == 1:
+        return True
+
+    return False
+
+
+def update_pr_comment(tarball, repo_name, pr_number, state, msg):
+    """Update PR comment which contains specific tarball name.
+
+    Args:
+        tarball (string): name of tarball that is looked for in a PR comment
+        repo_name (string): name of the repository (USER_ORG/REPOSITORY)
+        pr_number (string): pull request number
+        state (string): state (upload) to be used in update
+        msg (string): msg (succeeded or failed) describing upload result
+    """
+    funcname = sys._getframe().f_code.co_name
+
+    gh = github.get_instance()
+    repo = gh.get_repo(repo_name)
+    pull_request = repo.get_pull(pr_number)
+
+    # TODO does this always return all comments?
+    comments = pull_request.get_issue_comments()
+    for comment in comments:
+        # NOTE
+        # adjust search string if format changed by event handler
+        # (separate process running eessi_bot_event_handler.py)
+        re_tarball = f".*{tarball}.*"
+        comment_match = re.search(re_tarball, comment.body)
+
+        if comment_match:
+            log(f"{funcname}(): found comment with id {comment.id}")
+
+            issue_comment = pull_request.get_issue_comment(int(comment.id))
+
+            dt = datetime.now(timezone.utc)
+            comment_update = (f"\n|{dt.strftime('%b %d %X %Z %Y')}|{state}|"
+                              f"transfer of `{tarball}` to S3 bucket {msg}|")
+
+            # append update to existing comment
+            issue_comment.edit(issue_comment.body + comment_update)
+
+            # leave for loop (only update one comment, because tarball
+            # should only be referenced in one comment)
+            break
```

Somehow we should report an error if *no* matching comment was found? That's definitely a sign of trouble...

---

Address through #111